microsoft-edge/devtools-guide-chromium/console/console-dom-interaction.md

```diff
@@ -5,290 +5,345 @@ author: MSEdgeTeam
 
-   ![Use map on $$ to filter information down to the bare minimum](./console-dom-interaction-images/console-dom-filter-link-data.png)
+1. Go to the webpage that you want to inspect. For example, open the [DevTools Console tool DOM interactions demo page](https://microsoftedge.github.io/Demos/devtools-console-dom-interactions/) in a new tab or window.
 
-   You aren't done yet; several links are internal links to the webpage or have empty text.
+1. Press **Ctrl+Shift+J** (Windows, Linux) or **Command+Option+J** (macOS).  The **Console** tool opens in DevTools, next to the webpage.
 
-1. Use the `filter` method to get rid of the internal links:
+1. Paste the following code into the **Console**, and then press **Enter**:
 
    ```javascript
-   $$('a').map(a => {
-      return {text: a.innerText, url: a.href}
-   }).filter(a => {
-      return a.text !== '' && !a.url.match('learn.microsoft.com')
-   })
+   console.table($$('img'), ['src','alt'])
    ```
 
-   ![Get the links that aren't empty and are external](./console-dom-interaction-images/console-dom-filter-out-empty-links.png)
+   The list of `<img>` elements on the webpage is displayed in the **Console**, formatted as a table, with each image's `src` and `alt` attributes as columns:
+
+   ![The console tool, showing the console.table() function and a table listing all images](./console-dom-interaction-images/images-in-table.png)
+
+
+###### Get all page headings and anchor URLs
+
+In this example, you get all the headings on the webpage that have an `id` attribute, and you generate a URL that points to each heading.
 
-   By manipulating the DOM through issuing statements in the **Console**, you can change these elements in the rendered webpage.
+1. Go to the webpage that you want to inspect. For example, open the [DevTools Console tool DOM interactions demo page](https://microsoftedge.github.io/Demos/devtools-console-dom-interactions/) in a new tab or window.
 
-1. For example, enter the following code, which adds a green border around all external links:
+1. Press **Ctrl+Shift+J** (Windows, Linux) or **Command+Option+J** (macOS).  The **Console** tool opens in DevTools, next to the webpage.
+
+1. Copy and paste the following code into the **Console**, and then press **Enter**:
 
    ```javascript
-   $$('a[href^="https://"]').forEach(
-      a => a.style.border = '1px solid green'
-   )
+   let headings = $$('main [id]').filter(el => {
+     return el.nodeName.startsWith('H');
+   }).map(el => {
+     return `${el.innerText}: ${document.location.href}#${el.id}\n`;
+   });
+   console.log(headings.join("\n"));
    ```
+   
+   The result is text that contains content for each heading followed by the full URL that points to it:
```

Comment:
```suggestion
   The result is text that contains content for each heading followed by the full URL that points to the section heading anchor:
```
Replace "it" pointer by specific item.
alts:
* that points to the section heading anchor:
* that points to the section heading:
* that points to the section: